src/tarantool.c

```diff
@@ -225,7 +225,7 @@ static int __tarantool_connect(tarantool_object *t_obj) {
 	TSRMLS_FETCH();
 	tarantool_connection *obj = t_obj->obj;
 	int status = SUCCESS;
-	long count = TARANTOOL_G(retry_count);
+	long count = TARANTOOL_G(retry_count) + 1;
```

Comment:
So in case of `retry_count` > 0 the real retry count will be `retry_count` + 1? I think it should be handled in some other way.